### PR TITLE
reverseproxy: active health check allows configurable health_passes and health_fails

### PR DESCRIPTION
**Pull request by ottenhoff**
To fix issue #6152 

Please let me know if this is directionally correct.

### CODE REVIEW

**Comment by mholt** [comment]
Thanks for the PR!

This is directionally correct, yes :smile: It's looking good actually. Does it work well for you?

**Comment by ottenhoff** [comment]
Thanks! The previous code didn't reset the counter after marking a Host (un)healthy. Now this is working well in my basic testing and matches behavior of other load balancers

**Review by mholt — COMMENTED** [review]
Gotcha, thanks for the update!

Did another quick pass and it's looking good but I have a few minor quibbles over naming and capitalization :upside_down_face:

File: modules/caddyhttp/reverseproxy/hosts.go [hunk 1]
```diff
@@ -136,8 +136,10 @@ func (u *Upstream) fillHost() {
 // Host is the basic, in-memory representation of the state of a remote host.
 // Its fields are accessed atomically and Host values must not be copied.
 type Host struct {
-	numRequests int64 // must be 64-bit aligned on 32-bit systems (see https://golang.org/pkg/sync/atomic/#pkg-note-BUG)
-	fails       int64
+	numRequests  int64 // must be 64-bit aligned on 32-bit systems (see https://golang.org/pkg/sync/atomic/#pkg-note-BUG)
+	fails        int64
+	healthPasses int64
+	healthFails  int64
```
[the remark this answers is not in the document]
Comment (ottenhoff) [reply modules/caddyhttp/reverseproxy/hosts.go:142]: yes much more clear! i will make the change.

File: modules/caddyhttp/reverseproxy/hosts.go [hunk 2]
```diff
@@ -150,6 +152,16 @@ func (h *Host) Fails() int {
 	return int(atomic.LoadInt64(&h.fails))
 }
 
+// HealthPasses returns the number of consecutive active health check passes with the upstream.
+func (h *Host) HealthPasses() int {
```
[the remark this answers is not in the document]
Comment (ottenhoff) [reply modules/caddyhttp/reverseproxy/hosts.go:156]: they do not need to be exported. change coming.

**Review by mholt — APPROVED** [review]
This looks good now. Thank you for the contribution!!

**Pull request merged**